src/core/main.c

```diff
@@ -2129,6 +2129,8 @@ static int invoke_main_loop(
                         manager_send_reloading(m);
                         manager_set_switching_root(m, true);
 
+                        dual_timestamp_now(m->timestamps + MANAGER_TIMESTAMP_SOFTREBOOT_START);
```

Comment:
hmm, is this really the right place to take the timestamp? 

I think it should probably be taken the moment the shutdown jobs are enqueued.

i.e. somewhere close to job_shutdown_magic(). 

Otherwise this will not cover the slow shutdown some units might need.